### ISSUE TITLE
Firefox 117 disabled the Tab Mix Plus 

### ISSUE THREAD

**Issue by DavidXanatos** [onemen/TabMixPlus#235]
Firefox 117 disabled the Tab Mix Plus, i have downgraded to 116 for the time being,
please advise how to get it to work with 117

**Comment by onemen** [comment]
https://github.com/xiaoxiaoflood/firefox-scripts/issues/265

**Comment by okimurin** [comment]
After updating to firefox117, it became unusable.
Once I deleted TMP, I couldn't even install it.
I get an error that this add-on is broken and cannot be installed.

https://github.com/xiaoxiaoflood/firefox-scripts/issues/265
I tried this and it didn't work.

**Comment by onemen** [comment]
Use these files and follow [this](https://github.com/xiaoxiaoflood/firefox-scripts#instructions) instructions.

[Firefox config and pref](https://github.com/onemen/TabMixPlus/files/12257459/fx-folder.zip)

[utils → I'm interested in both scripts and extensions](https://github.com/onemen/TabMixPlus/files/12257460/utils_extensions_and_scripts.zip)
[utils → I'm only interested in extensions](https://github.com/onemen/TabMixPlus/files/12257461/utils_extensions_only.zip)
[utils → I'm only interested in scripts](https://github.com/onemen/TabMixPlus/files/12257458/utils_scripts_only.zip)

**Comment by okimurin** [comment]
I overwrote and rebooted, but could not install.

**Comment by markus88at** [comment]
I unfortunately have to second this, the update 117.0b3 fully broke it, even in the Developer Edition which I was using.

**Comment by onemen** [comment]
You need to unistall Tab mix
Install the config script https://github.com/onemen/TabMixPlus/files/12257459/fx-folder.zip

And install utils files from above

Donload an install Tab mix

**Comment by okimurin** [comment]
I overwrote the "utils" file in the Chrome folder in the profile folder, overwrote the contents of "fx-folder" in the installation folder, and restarted, but I could not install it.

**Comment by okimurin** [comment]
When I placed the "Config.zip" file posted by Mr. gelirhil in thread #233, I was able to install it successfully. TNX!

**Comment by DoulosTrieste** [comment]
I pasted the config from #233 but still could not install TMP on 117 Release.
Everything else already setup as per instructions from UserChromeJS.

**Comment by denmorgenshtern** [comment]
I've encountered the problem "This add-on could not be installed because it appears to be corrupt." 
Reinstalling the add-on did not work. 
Deleting it did not help either. 
Reinstalling all associated scripts and the "115 update" configuration did not work either. 

The issue was only resolved when I deleted the old associated files in all profiles (I have 2) folders and the application folder, and replaced them with new ones (although it seems that they are the same). With "Clear startup cache" of course.

**Comment by onemen** [comment]
latest Tab mix is [v1.0.0-pre.16](https://github.com/onemen/TabMixPlus/releases/tag/v1.0.0-pre.16)

latest Firefox is version 117

**Comment by VLM-TechEd** [comment]
Thank you. Still confused about what else to do; there appears to be no conclusion to this thread.

**Comment by onemen** [comment]
https://github.com/onemen/TabMixPlus/files/12257459/fx-folder.zip

[utils → I'm interested in both scripts and extensions](https://github.com/onemen/TabMixPlus/files/12257460/utils_extensions_and_scripts.zip)


install the latest scripts

**Comment by DoulosTrieste** [comment]
> [DoulosTrieste] Those are old files, are they not? And at this point, what version of TMP are we supposed to be installing? I have 117 (downloaded a while ago), but on the downloads/releases page it looks like 116 is the most recent version. I am confused.

They are the newest files, packed into 1 folder for convenience. I found config.js particularly troublesome, as if Windows 11 keeps restoring the old version from somewhere. But finally I got it to work. Use the newest TMP dev version.

**Comment by VLM-TechEd** [comment]
“Add-on could not be installed because it appears to be corrupt” until I remembered to clear cache and restart Fx. Now, everything works fine again. Thank you!

[DoulosTrieste], thank you, too!

**Comment by VLM-TechEd** [comment]
Oops, now the problem of the dialog running off the screen with high zoom again. You made a pre pre-pre-release version 17 for me which solved the problem. But I didn’t save it because I assumed it would be absorbed into a future release. The future release just hasn’t arrived yet.

**Comment by dxzy** [comment]
open my PC, autoupdate to 117, TMP dead too.
After trying many times, userChromeJS cannot be started, so the TMP installation is always broken.

**Comment by onemen** [comment]
> Oops, now the problem of the dialog running off the screen with high zoom again. You made a pre pre-pre-release version 17 for me which solved the problem. But I didn’t save it because I assumed it would be absorbed into a future release. The future release just hasn’t arrived yet.

latest test build is here:
https://tabmixplus.org/dev-builds/test%20builds/

**Comment by onemen** [comment]
> open my PC, autoupdate to 117, TMP dead too. After trying many times, userChromeJS cannot be started, so the TMP installation is always broken.
> 
> I don’t know why utils and config.js are always invalid. I gave up, tried everything above but still failed, beyond my capabilities.

![firefox](https://github.com/onemen/TabMixPlus/assets/3650909/e39c4d4e-5bec-47fe-96d7-faba7fab24b2)

1. Install latest scripts to your Application Binary folder, [config scripts](https://github.com/onemen/TabMixPlus/files/12257459/fx-folder.zip).
2. Profile Folder is the path to create chrome folder and install [utils](https://github.com/onemen/TabMixPlus/files/12257460/utils_extensions_and_scripts.zip) folder in it.
3. Remove Tab mix extension.
4. Open about:support and click "Clear startup cache…" to force Firefox to load userChromeJS on the next startup.
5. Start Firefox again.
6. Install Tab mix again.

**Comment by dxzy** [comment]
>     1. 將最新的腳本安裝到您​​的應用程序二進製文件夾， [配置腳本 ](https://github.com/onemen/TabMixPlus/files/12257459/fx-folder.zip)。
>     2. Profile文件夾是創建chrome文件夾並 [安裝utils文件夾的路徑。 ](https://github.com/onemen/TabMixPlus/files/12257460/utils_extensions_and_scripts.zip)在其中
>     3. 刪除選項卡混合擴展。
>     4. 打開 about:support 並單擊“清除啟動緩存...”以強制 Firefox 在下次啟動時加載 userChromeJS。
>     5. 再次啟動火狐瀏覽器。
>     6. 再次安裝 Tab mix。
Same steps, but it worked this time. Still not sure why, thank you.

**Comment by Vallek** [comment]
Make sure ''extensions.experiments.enabled' is enabled in about:config! Firefox 117 turned it off for me.

**Comment by VLM-TechEd** [comment]
>> Make sure ''extensions.experiments.enabled' is enabled in about:config! Firefox 117 turned it off for me.

Good point, thank you!

**Comment by VLM-TechEd** [comment]
latest test build is here:
https://tabmixplus.org/dev-builds/test%20builds/

Works perfectly, as before. Thank you!

**Comment by Brenex** [comment]
Thanks, re-enabling extensions.experiments.enabled (grrr) and then https://github.com/onemen/TabMixPlus/issues/235#issuecomment-1702163295 fixed it for me on 117.

**Comment by spartaque12** [comment]
this is such a nonsense ive tried half of day, and still cannot figure out why i cannot install extension, it always says
**this addon could not be installed because it appears to be corrupt firefox**
of course i did in about:config
**xpinstall.signatures.required = false
extensions.experiments.enabled = true**

**Comment by outdoor-miner** [comment]
> this is such a nonsense ive tried half of day, and still cannot figure out why i cannot install extension, it always says **this addon could not be installed because it appears to be corrupt** of course i did in about:config **xpinstall.signatures.required = false extensions.experiments.enabled = true**

This is probably a stupid question, but in step 2, did you put the files in a folder named "chrome" inside your profile folder? I missed that part a few times before I realized it and now it works like a charm!

**Comment by Qaezar** [comment]
> > open my PC, autoupdate to 117, TMP dead too. After trying many times, userChromeJS cannot be started, so the TMP installation is always broken.
> > I don’t know why utils and config.js are always invalid. I gave up, tried everything above but still failed, beyond my capabilities.
> 
> ![firefox](https://user-images.githubusercontent.com/3650909/263444507-e39c4d4e-5bec-47fe-96d7-faba7fab24b2.png)
> 
>     1. Install latest scripts to your Application Binary folder, [config scripts](https://github.com/onemen/TabMixPlus/files/12257459/fx-folder.zip).
> 
>     2. Profile Folder is the path to create chrome folder and install [utils](https://github.com/onemen/TabMixPlus/files/12257460/utils_extensions_and_scripts.zip) folder in it.
> 
>     3. Remove Tab mix extension.
> 
>     4. Open about:support and click "Clear startup cache…" to force Firefox to load userChromeJS on the next startup.
> 
>     5. Start Firefox again.
> 
>     6. Install Tab mix again.

Thank you so much for this one !

**Comment by spartaque12** [comment]
> > this is such a nonsense ive tried half of day, and still cannot figure out why i cannot install extension, it always says **this addon could not be installed because it appears to be corrupt** of course i did in about:config **xpinstall.signatures.required = false extensions.experiments.enabled = true**
> 
> This is probably a stupid question, but in step 2, did you put the files in a folder named "chrome" inside your profile folder? I missed that part a few times before I realized it and now it works like a charm!

nah, it used to work, but now i cannot make it work. I just dont understand why extension cannot be installed, it always says me that it's corrupt

**Comment by lonegungirl** [comment]
I kept getting a message that it wouldn't install the extension because it was corrupt, but then I repeated all the steps and now the message says that Firefox won't install the extension because it's unvalidated.

**Comment by onemen** [comment]
try this:
go to `about:config` and set `extensions.experiments.enabled` to `true`

**Comment by lonegungirl** [comment]
Extensions.experiments.enabled is still set to true from when it was working before this last update yesterday.

**Comment by VLM-TechEd** [comment]
Lonegungirl, when you finished setting things up, did you remember to clear cache? That did it for me.

**Comment by lonegungirl** [comment]
Yeah, cleared startup and all other caches I could find, but still nothing.  I tried installing Nightly to see if I could switch to that, but having run through all the instructions for that, I get the corrupt popup on that one, and the unvalidated popup on regular Firefox.

**Comment by Ramel** [comment]
> I kept getting a message that it wouldn't install the extension because it was corrupt, but then I repeated all the steps and now the message says that Firefox won't install the extension because it's unvalidated.

It seems we need to install: utils_extensions_and_scripts.zip

[utils → I'm interested in both scripts and extensions](https://github.com/onemen/TabMixPlus/files/12257460/utils_extensions_and_scripts.zip)

**Comment by pyrates999** [comment]
See here for how to install with firefox 117: https://github.com/onemen/TabMixPlus/discussions/237

**Comment by onemen** [comment]
version [1.0.0-pre.18](https://github.com/onemen/TabMixPlus/releases/tag/v1.0.0-pre.18) is out

**Comment by Kalinette** [comment]
Hello.

Unable to install TabMixPlus under freshly installed Firefox 122 or 117. I follow the procedures to the letter but I always get the answer:
This add-on could not be installed because it appears to be corrupt.

I've been on the problem for two days with no solutions.

**Comment by onemen** [comment]
@Kalinette,
Follow the instructions from here: [installation](https://github.com/onemen/TabMixPlus?tab=readme-ov-file#installation), [troubleshooting](https://github.com/onemen/TabMixPlus?tab=readme-ov-file#troubleshooting)
